commands/operator-sdk/cmd/generate/crd.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2018 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package generate
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	cmdError "github.com/operator-framework/operator-sdk/commands/operator-sdk/error"
+	"github.com/operator-framework/operator-sdk/pkg/generator"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	apiVersion string
+	kind       string
+)
+
+const (
+	goDir        = "GOPATH"
+	deployCrdDir = "deploy"
+)
+
+func NewGenerateCrdCmd() *cobra.Command {
+	crdCmd := &cobra.Command{
+		Use:   "crd",
+		Short: "Generates a custom resource definition (CRD) and the custom resource (CR) files",
+		Long: `The operator-sdk generate command will create a custom resource definition (CRD) and the custom resource (CR) files for the specified api-version and kind.
+
+Generated CRD filename: <project-name>/deploy/<kind>_crd.yaml
```

Comment:
@shawn-hurley Just wanted to point out that if you try to generate a second CRD with the same Kind but different APIVersion then that will overwrite the previous `<kind>_crd.yaml`. Not sure if that's a legitimate use case in an ansible type project but we account for multiple CRDs in the refactor branch by using the format `deploy/crds/<group>_<version>_<kind>_crd.yaml`

https://github.com/operator-framework/operator-sdk/blob/refactor/controller-runtime/commands/operator-sdk/cmd/add/api.go#L146-L168

We'll soon add the changes from this PR into the refactor branch as well(to fit the style of the scaffold). 
Right now we don't have a separate `generate CRD` command but I'm wondering if you wanted to make the CRD generation consistent with how we do it in the refactor branch.

Reply:
This makes sense to me... @johnkim76 any concerns around this?

Reply:
that sounds good.  I'll make the update to the filename. thank you.